lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -81,19 +107,45 @@ public function get_providers() {
 	public function register_font( $font ) {
 		$font = $this->validate_font( $font );
 		if ( $font ) {
-			$id                    = $this->get_font_id( $font );
-			$this->webfonts[ $id ] = $font;
+			$slug = $this->get_font_slug( $font );
+
+			if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
+				$this->registered_webfonts[ $slug ] = array();
+			}
+
+			$this->registered_webfonts[ $slug ][] = $font;
+		}
+	}
+
+	public function enqueue_font( $font_family ) {
+		$slug = $this->get_font_slug( $font_family );
+
+		if ( isset( $this->enqueued_webfonts[ $slug ] ) ) {
+			_doing_it_wrong( __FUNCTION__, sprintf( __( 'The "%s" font family is already enqueued.' ), $slug ), '6.0.0' );
+			return false;
+		}
+
+		if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
+			_doing_it_wrong( __FUNCTION__, sprintf( __( 'The "%s" font family is not registered.' ), $slug ), '6.0.0' );
+			return false;
 		}
```

Comment:
What about this instance of `_doing_it_wrong()`?  Does this qualify as an egregious instance? Yes, I think so as this function should never be called if the webfont is not registered with the API. BUT having `trigger_error()` above and `_doing_it_wrong()` here will be confusing and less readable. Contributors will read the code and stop to say "Huh". Not good. I think instead, using `trigger_error()` here is likely better for readability and consistency within this class.

```suggestion
			trigger_error( 
				sprintf(
					/* translators: %s unique slug to identify the webfont */
					__( 'The "%s" font family is not registered.', 'gutenberg' ),
					$slug
				)
			);
```